sensor/common/scan/scan.go

```diff
@@ -20,64 +24,103 @@ var (
+	if features.ImageSignatureVerification.Enabled() {
+		sigs, err = fetchSignaturesWithRetry(ctx, signatures.NewSignatureFetcher(), image,
+			matchingRegistry)
+		if err != nil {
+			log.Debugf("Failed fetching signatures for image %q: %v", imgName, err)
+			return nil, errors.Wrapf(err, "fetching signature for image %q from registry %q",
+				imgName, matchingRegistry.Name())
+		}
 	}
 
-	// Image ID may not exist, if this image is not from an active deployment
-	// (for example the Admission Controller checks the image prior to deployment).
-	// Try to determine the SHA with best-effort.
-	sha := utils.GetSHAFromIDAndMetadata(image.GetId(), metadata)
+	// Retrieve the image ID with best-effort from image and metadata.
+	imgID := utils.GetSHAFromIDAndMetadata(image.GetId(), metadata)
 
-	// Get the image's vulnerabilities from Central.
-	centralResp, err := centralClient.GetImageVulnerabilitiesInternal(ctx, &v1.GetImageVulnerabilitiesInternalRequest{
-		ImageId:        sha,
+	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
+	// signature verification results.
+	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
+		ImageId:        imgID,
 		ImageName:      image.GetName(),
 		Metadata:       metadata,
-		IsClusterLocal: image.GetIsClusterLocal(),
-		Components:     scanResp.GetComponents(),
-		Notes:          scanResp.GetNotes(),
+		Components:     scannerResp.GetComponents(),
+		Notes:          scannerResp.GetNotes(),
+		ImageSignature: &storage.ImageSignature{Signatures: sigs},
 	})
 	if err != nil {
-		log.Debugf("Unable to retrieve image vulnerabilities for %s: %v", name, err)
-		return nil, errors.Wrapf(err, "retrieving image vulnerabilities for %s", name)
+		log.Debugf("Unable to enrich image %q: %v", imgName, err)
+		return nil, errors.Wrapf(err, "enriching image %q via central", imgName)
 	}
-	log.Debugf("Retrieved image vulnerabilities for %s", name)
 
-	// Return the completely scanned image.
+	log.Debugf("Retrieved image enrichment results for %q", imgName)
+
 	return centralResp.GetImage(), nil
 }
+
+// scanImage will scan the given image and return its components.
+// It will return ErrNoLocalScanner if no local scanner is available. It will return any errors that occurred during
+// receiving scan results from local scanner or if the scan status was non-successful.
+func scanImage(ctx context.Context, image *storage.Image,
+	registry registryTypes.Registry, scannerClient *scannerclient.Client) (*scannerV1.GetImageComponentsResponse, error) {
+	// Get the image analysis from the local Scanner.
+	scanResp, err := scannerClient.GetImageAnalysis(ctx, image, registry.Config())
+	if err != nil {
+		return nil, err
+	}
+	// Return an error indicating a non-successful scan result.
+	if scanResp.GetStatus() != scannerV1.ScanStatus_SUCCEEDED {
+		return nil, fmt.Errorf("scan failed with status %q", scanResp.GetStatus().String())
```

Comment:
We don't receive an error from `scannerClient.GetImageAnalysis`, hence `fmt.Errorf`.
Currently, I think you used `errors.Wrapf`, but this would return nil, since `err` would be nil as well.